Assets/Scripts/Traps/ProyectileTrap.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ProyectileTrap : MonoBehaviour
+{
+
+    [field: SerializeField] public float Damage { get; private set; }
+    [field: SerializeField] public float Speed { get; private set; }
+    [field: SerializeField] public float LifeTime { get; private set; }
+
+    [SerializeField] bool _shootReady;
+    [SerializeField] float _shootQuantity;
+
+    [Header("References")]
+
+    [SerializeField] Transform _bulletSpawnPoint;
+
+    [SerializeField] GameObject _proyectile;
+
+
+
+    public void CheckShoot()
```

Comment:
We don't use coroutines in the project, but I think it's ok because you don't know how to use async methods.

One thing here I'd change - rename this method to just `Shoot` and the coroutine method to `ShootCoroutine`. Because imagine you don't know how this class is implemented and when you call `CheckShoot` from other classes you will wonder "why is it called `Check..`?" For a user of class it's more logical to call a simple method `Shoot` because you shouldn't care how it's implemented internally.

Reply:
Renamed this method to `ShootCoroutine` as suggested.

Reply:
You did everything the other way around :) The public method should be called `Shoot` and the private one is `ShootCoroutine` because it's internal logic.